### ISSUE TITLE
Image digest result is <none>

### ISSUE THREAD

**Issue by northtyphoon** [Azure/acr-builder#52]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG

**What happened?**:
Unable to get image digests after build succeeds. All values are \<none\>.

**What did you expect to happen?**:
Valid digest value for each image.

**How do you reproduce it (as minimally and precisely as possible)?**:
1. Run ```docker build --rm -t acr-builder .``` to create the acr-builder container image
2. Run ```./scripts/run-build.sh``` (The script basically uses acr-builder container to build acr-builder source code.)
3. Error: Running command docker image ls --digests --format "{{.Digest}}" acr-builder 
"\<none\>"

**Anything else we need to know?**:
The issue seems related to [https://github.com/docker/cli/issues/728](https://github.com/docker/cli/issues/728). 
A potential fix is to move image digest evaluation task after push.

**Environment**:
Ubuntu 16.4 + Docker 17.12.0-ce

@shhsu

**Comment by shhsu** [comment]
Yes, pushing before evaluating the digests would solve the issue

**Comment by northtyphoon** [comment]
So we found two problems
1. For output image, we need to push it first to get the digest.
2. For dependency image, the command like ```docker image ls --digest docker:17.12.0-ce-git``` is unable to get the image digest, only \<none\>. It's probably related to docker/cli#728.

**Comment by northtyphoon** [comment]
We found another issue. 

For multi-arch docker images (eg, [Ubuntu](https://hub.docker.com/_/ubuntu/)), ```docker inspect``` or ```docker image``` only returns the __master digest__ for all platform. Ideally, we want to provide the specific image digest for the current platform for dependency tracking.

For now, we will just provide the  __master digest__. Once the toolchain is ready, we can switch to platform-specific digest.

cc @djyou
